Qpf/Macro/Data/Constructors.lean

```diff
@@ -13,11 +13,7 @@ partial def countConstructorArgs : Syntax → Nat
   | Syntax.node _ ``Term.arrow #[_, _, tail]  =>  1 + (countConstructorArgs tail)
   | _                                         => 0
 
-open Elab
-/--
-  Add convenient constructor functions to the environment
--/
-def mkConstructors (view : DataView) (shape : Name) : CommandElabM Unit := do
+def mkConstructorsWithNameAndType (view : DataView) (shape : Name) (nameGen : CtorView → Ident) (ty : Term) := do
```

Comment:
```suggestion
def mkConstructorsWithNameAndType (view : DataView) (shape : Name) (nameGen : CtorView → Name) (ty : Term) := do
```
I feel like having the argument generate a `Name` is cleaner than making it generate `Ident`s